### PR TITLE
Use kinder to replace minikube

### PR DESCRIPTION
**Pull request by namnx228**

### CODE REVIEW

**Comment by Jaakko-Os** [comment]
Commit message could be rewritten. Now it spreads to multiple lines and several empty lines.

File: 01_install_requirements.sh [hunk 1]
```diff
@@ -1,23 +1,60 @@
...
   source centos_install_requirements.sh
 fi
 
-if ! which minikube 2>/dev/null ; then
-    curl -Lo minikube https://storage.googleapis.com/minikube/releases/latest/minikube-linux-amd64
-    chmod +x minikube
-    sudo mv minikube /usr/local/bin/.
-fi
+# Create Go directory
+mkdir -p ${HOME}/go/{bin,src,pkg/mod}
```
Comment (maelk) [inline 01_install_requirements.sh]: rather use the GOPATH variable, moving the go env call if needed

File: vm-setup/libvirt-package-playbook.yml [hunk 2]
```diff
@@ -0,0 +1,11 @@
+---
```
Comment (maelk) [inline vm-setup/libvirt-package-playbook.yml]: remove, unused

File: vm-setup/vars/package_installation.yml [hunk 3]
```diff
@@ -0,0 +1,95 @@
+packages_installation:
```
Comment (maelk) [inline vm-setup/vars/package_installation.yml]: the file name should maybe be packageS too ?

File: vm-setup/roles/packages_installation/tasks/main.yml [hunk 4]
```diff
@@ -20,17 +20,17 @@
...
           when: ansible_lsb.major_release|int == 7
           package:
-             name: "{{ install.centos.package +  install.centos.libvirt.rhel7 }}"
+             name: "{{ package.centos.package +  package.centos.libvirt.rhel7 }}"
              state: present
         - name: Install packages on rhel8
           when: ansible_lsb.major_release|int == 8
           package:
-             name: "{{ install.centos.package + install.centos.libvirt.rhel7 }}"
+             name: "{{ package.centos.package + package.centos.libvirt.rhel7 }}"
```
Comment (maelk) [inline vm-setup/roles/packages_installation/tasks/main.yml]: this needs to be changed to rhel8

File: vm-setup/install-package-playbook.yml [hunk 5]
```diff
@@ -4,12 +4,12 @@
   connection: local
   gather_facts: true
   vars_files:
-    - vars/input.yml
+    - vars/package_installation.yml
```
Comment (maelk) [inline vm-setup/install-package-playbook.yml]: packages_... rather ?

File: README.md [hunk 6]
```diff
@@ -32,7 +32,7 @@ The `Makefile` runs a series of scripts, described here:
 * `02_configure_host.sh` - Create a set of VMs that will be managed as if they
   were bare metal hosts.
 
-* `03_launch_mgmt_cluster.sh` - Launch a management cluster using `minikube` and
+* `03_launch_mgmt_cluster.sh` - Launch a management cluster using `kinder` and
```
Comment (maelk) [inline README.md]: minikube or kinder

File: 03_launch_mgmt_cluster.sh [hunk 7]
```diff
@@ -59,6 +59,23 @@ function launch_minikube() {
     minikube start
 }
 
+function launch_kinder() {
+  kinder="${GOPATH}/bin/kinder"
+  if [[ $(sudo $kinder get clusters) == "" ]]; then
+    sudo $kinder create cluster
+    sudo $kinder do kubeadm-init
```
Comment (maelk) [inline 03_launch_mgmt_cluster.sh:72]: the cluster won't have an interface in the provisioning network, that is problematic if for example we move ironic inside the cluster

**Comment by nordixinfra** [comment]
Can one of the admins verify this patch?

File: lib/common.sh [hunk 8]
```diff
@@ -33,6 +34,13 @@ INT_IF=${INT_IF:-}
 #Root disk to deploy coreOS - use /dev/sda on BM
 ROOT_DISK_NAME=${ROOT_DISK_NAME-"/dev/sda"}
 #Container runtime
+
+K8S=${K8S:-"minikube"}
```
Comment (maelk) [inline lib/common.sh:37]: Please add this in config_example.sh

File: vm-setup/roles/libvirt/tasks/install_package_tasks.yml [hunk 9]
```diff
@@ -0,0 +1,27 @@
+# Install the packages required for our desired libvirt environment.
```
Comment (maelk) [inline vm-setup/roles/libvirt/tasks/install_package_tasks.yml]: this should not be in this PR

File: lib/common.sh (hunk 8 above)
Comment (namnx228) [reply lib/common.sh:37]: Done

**Comment by maelk** [comment]
add to whitelist

**Comment by maelk** [comment]
we will not take this further for now

**Pull request closed without merging**